packages/components/inputs/select-utils/src/create-select-styles.js

```diff
@@ -235,6 +235,10 @@ const valueContainerStyles = (props, theme) => (base) => {
     padding: '0',
     backgroundColor: 'none',
     overflow: 'hidden',
+    display:
+      (props.iconLeft && !props.isMulti) || (props.isMulti && props.hasValue)
+        ? 'flex'
+        : 'grid',
```

Comment:
Remember to leave a comment describing the reason for this logic (and also maybe a link to the react-select original PR).

Reply:
done.